packages/labs/context/src/lib/controllers/context-provider.ts

```diff
@@ -56,7 +56,7 @@ export class ContextProvider<T extends ContextKey<unknown, unknown>>
   public onContextRequest = (
     ev: ContextRequestEvent<ContextKey<unknown, unknown>>
   ): void => {
-    if (ev.context !== this.context) {
+    if (ev.context !== this.context || ev.composedPath()[0] === this.host) {
```

Comment:
I think we can check `ev.target === this.host`, since we're guarding against the type of dispatch call on line 72. I think it's a little more clear too.

Reply:
@justinfagnani - I wish you are correct. 

`ev.target === this.host` was my initial - non working - code. `ev.target === this.host` is alway truthy in my case, even when the event is dispatched from deeper in the shadowDom hierarchy. 

The thing is, I don't understand why it seems the event is retargeted when it should obviously not. 

I'll try to provide a test to ensure the correct behavior

Reply:
Actrually, I think we need `ev.composedPath()[0] === this.host`. 

Consider the case where the consumer is in the shadow dom of the supplier (for the same context). By the time the event crosses the shadow boundary, it is retargetted and hence `ev.target === this.host` is truly (and the context would not be  supplied to the consumer). Here, on the contrary we want to the consumer to get the context.

Reply:
My understanding of retargetting is spotty, but this sounds right to me?

Would calling `composedPath()` be a performance issue @justinfagnani?

Reply:
Ah, yes, All composed events are retargeted. This shouldn't be a perf problem, and is limited to host that consume and provide the same context, so it's not called on all events.